docs/DeveloperGuide.md

```diff
@@ -1,38 +1,275 @@
+
+Step 3. Since egg is part of our list of known ingredients, a new `Ingredient` object would be created with the quantity, 5, and name, egg, specified by the user and added to `IngredientList` with the `add()` method.
+
+Below shows the sequence diagram for the above AddCommand:
+
+![AddIngredientSequenceUML.png](images/AddIngredientSequenceUML.png)
+
+Continuing from the above AddCommand:
+
+Step 4. Now the user decides to remove 2 eggs and inputs his command e.g. `remove egg /qty 2`, the `IngredientList` would be searched through to see if egg is in the list
+
+Step 5. Since egg is in the list, its corresponding index in the list would be generated. 
+
+Step 6. The current quantity of egg in the list would be obtained by calling `getQuantity()` and checked to see if it is greater or equals to '2' the quantity input by the user
+
+Step 7. Since the quantity input by the user is smaller than the current quantity of egg, which is 3, in the `IngredientList`, the new quantity would be calculated to be 3 and updated by calling `setQuantity(3)` 
+
+Below shows the sequence diagram for the above RemoveCommand:
+
+![RemoveIngredientSequenceUML.png](images/RemoveIngredientSequenceUML.png)
+
+###### [Back to table of contents](#table-of-contents)
+
+### Storage Feature
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+The proposed storage mechanism of ingredients is facilitated by `IngredientStorage`. 
 
+It requires `IngredientList` of `MealCompanionSession`.
 
-## Product scope
-### Target user profile
+This features saves ingredient list data in json format and reads them back into ingredient objects.
 
-{Describe the target user profile}
+The following sequence diagram shows how data storage works:
 
-### Value proposition
+![StorageSequenceUML.png](images/StorageSequenceUML.png)
 
-{Describe the value proposition: what problem does it solve?}
+###### [Back to table of contents](#table-of-contents)
 
-## User Stories
+### Recipe Detail Command
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+The recipe command is facilitated by `RecipeDetailCommand`. 
+
+It requires `RecipeList` of `MealCompanionSession`. 
+
+The recipe commands takes in either a recipe index or a recipe name as parameter. The latter is resolved into the recipe's corresponding index number.
+
+The `Recipe` is retrieved from `RecipeList` and its details are outputted.
+
+The following sequence diagram shows how the Recipe Detail Command works:
+
+![RecipeDetailCommandSequence.png](images%2FRecipeDetailCommandSequence.png)
```

Comment:
![Screenshot from 2023-03-31 13-18-58](https://user-images.githubusercontent.com/2891116/229029958-ce523bb6-d22f-4481-8e2b-bc1248e3f37f.png)

The activation arrow for getingredients() is directed to the recipe instead of recipes:RecipeList.

:RecipeDetailCommand is also not deactivated at the end.